### PR TITLE
fix: truncate strings with mb_substr instead of substr

### PR DESCRIPTION
**Pull request by chuckadams**
# Pull Request

## What changed?

When populating plugins, fields are truncated with `mb_substr` instead of `substr`

## Why did it change?

Using `substr` cuts off UTF-8 byte sequences in the middle, which makes postgres very sad

## Did you fix any specific issues?

none

## CERTIFICATION

By opening this pull request, I do agree to abide by
the [CODE OF CONDUCT](https://github.com/aspirepress/.github/CODE_OF_CONDUCT.md) and be bound by the terms
of the [Contribution Guidelines](https://github.com/aspirepress/.github/CONTRIBUTING.md) in effect on the date and time
of my contribution as proven by the
revision information in GitHub. I also agree that any previous contributions shall be deemed subject to the terms of the
version in effect on the date and time of this pull request, or any future revisions for pull requests I may submit.
Further, I certify that this work is my own, is original, does not violate the intellectual property of any other person
or entity, and I am not violating any license agreements or contracts I have with any person or entity. Finally, I agree
that this code may be licensed under any license deemed appropraite by AspirePress, including but not
limited to open source, closed source, proprietary or custom licenses, and that such license terms neither violate my
rights or my copyright to this code.

### CODE REVIEW

File: app/Models/WpOrg/Plugin.php
Comment (asirota) [inline app/Models/WpOrg/Plugin.php:1]: From my experience mb_substr is not always install on all hosts. Can you add a check to make sure the module is loaded and throw an error to install it. Or   fall back to substr instead.

**Review by asirota — CHANGES_REQUESTED** [review]

File: app/Models/WpOrg/Plugin.php
Comment (chuckadams) [reply app/Models/WpOrg/Plugin.php:1]: I added it to composer's require.json.  falling back to the wrong function is not an option.

Comment (chuckadams) [reply app/Models/WpOrg/Plugin.php:1]: Also, this runs in a docker container where the relevant extensions are guaranteed to be installed

**Review by asirota — APPROVED** [review]

**Comment by asirota** [comment]
@chuckadams does the system give a warning when mb_string extension is not installed in php? I have seen this dependency be missing on many php installs especially in hosting.

**Comment by chuckadams** [comment]
The `ext-mbstring` requirement is in composer.json, so the whole thing will fail to build if it's missing.  IMHO any PHP install missing mbstring these days is a HABS of PHP (Hacked Up Bastardized Simalucra)

**Comment by asirota** [comment]
My https://newpathconsulting.com/wap plugin has a specific check for this. It's crazy how many hosts turn of mb_string out of sheer ignorance or foolish fear.

**Pull request merged**